.github/workflows/_torchprime_ci.yml

```diff
@@ -0,0 +1,107 @@
+      # section of the repo.
+      PERSONAL_ACCESS_TOKEN_FOR_TRIGGERING_TORCHPRIME:
+        required: true
+      GCLOUD_SERVICE_KEY:
+        required: true
+jobs:
+  torchprime-e2e-test:
+    name: Run torchprime E2E tests
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Use Docker in rootless mode
+        if: inputs.has_code_changes == 'true'
+        uses: ScribeMD/rootless-docker@0.2.2
+      - name: Add user to docker group
+        if: inputs.has_code_changes == 'true'
+        run: |
+          sudo usermod -aG docker $USER
+          newgrp docker
+        shell: bash
+      # Googlers: if this fails, follow http://shortn/_61iSj31q1b to debug.
+      - uses: google-github-actions/auth@v2
+        if: inputs.has_code_changes == 'true'
+        with:
+          credentials_json: '${{ secrets.GCLOUD_SERVICE_KEY }}'
+      - uses: google-github-actions/setup-gcloud@v2
+        if: inputs.has_code_changes == 'true'
+        with:
+          version: '>= 363.0.0'
+          install_components: 'beta,gke-gcloud-auth-plugin'
+      - name: Verify GCP setup
+        if: inputs.has_code_changes == 'true'
+        run: gcloud info
+        shell: bash
+      - name: Authenticate Docker
+        if: inputs.has_code_changes == 'true'
+        run: gcloud auth configure-docker --quiet
+        shell: bash
+      - name: Activate SA credentials
+        if: inputs.has_code_changes == 'true'
+        run: gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+        shell: bash
+      - name: Checkout infra
+        if: inputs.has_code_changes == 'true'
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            infra
+          fetch-depth: 1
+          path: pytorch-xla
+      # Build a docker image for torchprime E2E test
+      # First download the torch-xla-wheels
+      - name: Fetch wheels
+        if: inputs.has_code_changes == 'true'
+        uses: actions/download-artifact@v4
+        with:
+          name: torch-xla-wheels
+          path: /tmp/wheels/
+      # Generate a random 16-character UUID for the docker tag
+      - name: Generate random UUID tag
+        if: inputs.has_code_changes == 'true'
+        id: random_tag
+        shell: bash
+        run: |
+          echo "uuid=$(openssl rand -hex 8)" >> $GITHUB_OUTPUT
+      # Then run docker to install them and push a docker
+      - name: Build and push docker image
+        if: inputs.has_code_changes == 'true'
+        shell: bash
+        working-directory: pytorch-xla
+        run: |
+          . ./infra/ansible/build_for_torchprime.sh
+        env:
+          DEFAULT_CONTEXT_PATH: /tmp/wheels
+          DOCKER_IMAGE_NAME: for-torchprime-ci
+          DOCKER_IMAGE_TAG: ${{ steps.random_tag.outputs.uuid }}
+          DOCKER_PROJECT: tpu-pytorch
+      # Trigger torchprime E2E test workflow
+      - uses: convictional/trigger-workflow-and-wait@v1.6.5
```

Comment:
Who provisions for this workflow? Does it have enough capacity for our needs?

Reply:
Based on my read of GitHub docs, using the `convictional/trigger-workflow-and-wait@v1.6.5` workflow just means we're taking this guy's code and then running it.

That means the costs of running this step is charged to the PyTorch/XLA repo, which, in case of public repo and standard runners, is free.

The costs of running the torchprime E2E test is charged to the torchprime repo, which, being open source and using the standard ubuntu runner, is also free.